lib/wallaroo/core/sink/connector_sink/connector_sink.pony

```diff
@@ -133,20 +132,41 @@ actor ConnectorSink is Sink
   var _reconnect_pause: U64
   var _host: String
   var _service: String
+  let _worker_name: WorkerName
+  let _protocol_version: String
+  let _cookie: String
+  let _auth: ApplyReleaseBackpressureAuth
   var _from: String
 
   // Producer (Resilience)
   let _timers: Timers = Timers
 
   var _seq_id: SeqId = 0
 
+  // 2PC
+  var _twopc_state: cp.TwoPCFsmState = cp.TwoPCFsmStart
+  var _twopc_txn_id: String = ""
+  var _twopc_txn_id_at_close: String = ""
+  var _twopc_barrier_token_initial: CheckpointBarrierToken =
+    CheckpointBarrierToken(0)
+  var _twopc_barrier_token: CheckpointBarrierToken =
+    _twopc_barrier_token_initial
+  var _twopc_barrier_token_at_close: CheckpointBarrierToken =
+    _twopc_barrier_token_initial
+  var _twopc_phase1_commit: Bool = false
+  var _twopc_last_offset: USize = 0
+  var _twopc_current_offset: USize = 0
+
   new create(sink_id: RoutingId, sink_name: String, event_log: EventLog,
     recovering: Bool, env: Env, encoder_wrapper: ConnectorEncoderWrapper,
     metrics_reporter: MetricsReporter iso,
-    barrier_initiator: BarrierInitiator, checkpoint_initiator: CheckpointInitiator,
-    host: String, service: String, initial_msgs: Array[Array[ByteSeq] val] val,
+    barrier_coordinator: BarrierCoordinator, checkpoint_initiator: CheckpointInitiator,
+    host: String, service: String, worker_name: WorkerName,
+    protocol_version: String, cookie: String,
+    auth: ApplyReleaseBackpressureAuth,
+    initial_msgs: Array[Array[ByteSeq] val] val,
     from: String = "", init_size: USize = 64, max_size: USize = 16384,
-    reconnect_pause: U64 = 10_000_000_000)
+    reconnect_pause: U64 = 500_000_000 /* TODO: 10_000_000_000 */)
```

Comment:
Why is this a TODO?  Are you only temporarily changing this value?

Reply:
The 10 second reconnect timeout is really inconvenient for testing.  I'll hack this to use an `ifdef debug` constant.

Reply:
Could this be exposed as an option passed in at init? I can imagine different apps requiring different settings -- your inconvenience shows that others will find it inconvenient as well.